### ISSUE TITLE
Standardize class __repr__'s

### ISSUE THREAD

**Issue by jlowin** [PrefectHQ/prefect#617]
As observed in https://github.com/PrefectHQ/prefect/pull/616#discussion_r253284001, class reprs are inconsistent:

Core objects have `<>` around them:
```
<Flow: name=Flow>
<Task: Task>
<Edge: Task to Task>
```

States are "called" types:

```
Success()
Failed()
```

Runners use Python's default:

```
<prefect.engine.flow_runner.FlowRunner at 0x10e8d6e80>
```

**Comment by nandahkrishna** [comment]
Is it alright if I work on this issue?

**Comment by cicdw** [comment]
Yea absolutely @nandahkrishna that'd be great! Let me know if you have any questions about which objects need updating, etc.

**Comment by nandahkrishna** [comment]
It would be great if you could give me an idea on which objects need updating. I just had a look at some of the class definitions and spotted a few (like Runner, which you have mentioned earlier).

**Comment by jlowin** [comment]
![](https://media.giphy.com/media/l0HlHFRbmaZtBRhXG/giphy.gif)

**Comment by cicdw** [comment]
Here's a list to get you started:
- all `State` objects in `src/prefect/engine/state.py` (you should only have to update the base `State` class `__repr__`)
- all `Runner` objects (`TaskRunner`, `FlowRunner`, `CloudTaskRunner`, `CloudFlowRunner`)
- all executors (`DaskExecutor` `LocalExecutor`, `SyncExecutor`)

I'll update here if I find any others or if @jlowin thinks of any that I'm missing.

**Comment by nandahkrishna** [comment]
Thanks! I'll get started on this.

**Comment by jlowin** [comment]
I have a suspicion that for each type of class, only the base class really needs updating -- `State`, `Runner`, `BaseExecutor`.

**Comment by nandahkrishna** [comment]
> I have a suspicion that for each type of class, only the base class really needs updating -- `State`, `Runner`, `BaseExecutor`.

I'll check first by updating the base classes alone, and then if required update the rest.

Thanks @cicdw for the updated list. And yes, I'll make a PR even if it has just a few changes - helps me when I should change stuff after you review it!

**Comment by nandahkrishna** [comment]
I've created a PR #1050 for this, and so far I've updated the base classes Runner and State, which are now represented as `<Runner>` and `<State()>`. Do tell me if the format/details in the `__repr__` have to be changed.

**Comment by cicdw** [comment]
Closed via #1050
